entity/volume_group.go

```diff
@@ -2,17 +2,44 @@ package entity
 
 // VolumeGroup represents the MAAS VolumeGroup endpoint.
 type VolumeGroup struct {
-	LogicalVolumes     interface{} `json:"logical_volumes,omitempty"`
-	Devices            interface{} `json:"devices,omitempty"`
-	HumanSize          string      `json:"human_size,omitempty"`
-	UUID               string      `json:"uuid,omitempty"`
-	HumanAvailableSize string      `json:"human_available_size,omitempty"`
-	SystemID           string      `json:"system_id,omitempty"`
-	ResourceURI        string      `json:"resource_uri,omitempty"`
-	HumanUsedSize      string      `json:"human_used_size,omitempty"`
-	Name               string      `json:"name,omitempty"`
-	Size               int64       `json:"size,omitempty"`
-	UsedSize           int64       `json:"used_size,omitempty"`
-	AvailableSize      int64       `json:"available_size,omitempty"`
-	ID                 int         `json:"id,omitempty"`
+	Devices            interface{}     `json:"devices,omitempty"`
+	ResourceURI        string          `json:"resource_uri,omitempty"`
+	HumanSize          string          `json:"human_size,omitempty"`
+	UUID               string          `json:"uuid,omitempty"`
+	HumanAvailableSize string          `json:"human_available_size,omitempty"`
+	SystemID           string          `json:"system_id,omitempty"`
+	HumanUsedSize      string          `json:"human_used_size,omitempty"`
+	Name               string          `json:"name,omitempty"`
+	LogicalVolumes     []LogicalVolume `json:"logical_volumes,omitempty"`
```

Comment:
Why not using `BlockDevice`? or even better create a `VirtualBlockDevice` which is missing from our entities and is an extension of a `BlockDevice`? [link-1](https://github.com/canonical/maas/blob/master/src/maasserver/api/volume_groups.py#L175) [link-2](https://github.com/canonical/maas/blob/master/src/maasserver/models/filesystemgroup.py#L871) [link-3](https://github.com/canonical/maas/blob/master/src/maasserver/models/virtualblockdevice.py#L65)

Reply:
I'm not sure if it is correct to use the `BlockDevice` type as is or to add `VirtualBlockDevice`. However using `BlockDevice` seems simpler and better at the moment, so I fixed it in [9464fad](https://github.com/canonical/gomaasclient/pull/94/commits/9464fadd3f0cbb0e9dc56e7606d52c180423e50e).

Reply:
Would it be more appropriate to define it as `VirtualBlockDevice`? I will make corrections if necessary.

Reply:
That would be ideal, yes. The only reason I proposed the `BlockDevice` was to not overwhelm you. You could give it a try and I would propose any missing piece afterwards.

Reply:
I've added `VirtualBlockDevice` as an extension to `BlockDevice` in 374a4df. Could you please confirm that this is what you intended?